common/thorhelper/thorfile.cpp

```diff
@@ -289,11 +289,18 @@ static void gatherDerivedIndexInformation(DerivedIndexInformation & result, IDis
     //The following will depend on the compression format - e.g. if compressed searching is implemented
     result.sizeMemoryBranches = result.sizeOriginalBranches;
 
-    //Could calculate from the size of the raw record (with blobids) multiplied by the count.
-    if ((result.numBlobNodes == 0) && result.sizeOriginalData)
-        result.sizeMemoryLeafs = result.sizeOriginalData;
-    else if (result.sizeOriginalData)
-        result.sizeMemoryLeafs = (offset_t)((double)result.numLeafNodes * result.sizeOriginalData)/(result.numLeafNodes + result.numBranchNodes);
+    //NOTE: sizeOriginalData now includes the blob sizes that areremoved before passing to the builder
+    //      if the original blob size is recorded then use it, otherwise estimate it
+    if (result.sizeOriginalData)
+    {
+        offset_t originalBlobSize = attrs.getPropInt64("@originalBlobSize");
+        if (result.numBlobNodes == 0)
+            result.sizeMemoryLeafs = result.sizeOriginalData;
+        else if (originalBlobSize)
+            result.sizeMemoryLeafs = result.sizeOriginalData - originalBlobSize;
```

Comment:
Not new, but shouldn't it be sizeMemoryLeaves ?

Reply:
It should be, most places I tried to avoid the plural because it looks a bit ugly.  I'll change though.